staging/src/k8s.io/apiserver/pkg/storage/etcd3/preflight/checks.go

```diff
@@ -59,9 +60,12 @@ func parseServerURI(serverURI string) (*url.URL, error) {
 // CheckEtcdServers will attempt to reach all etcd servers once. If any
 // can be reached, return true.
 func (con EtcdConnection) CheckEtcdServers() (done bool, err error) {
-	// Attempt to reach every Etcd server in order
-	for _, serverURI := range con.ServerList {
-		host, err := parseServerURI(serverURI)
+	rand.Seed(time.Now().UTC().UnixNano())
+	// Attempt to reach every Etcd server randomly.
+	serverNumber := len(con.ServerList)
```

Comment:
Need to handle the edge case where con.ServerList has len of 0

Reply:
Yes, fixed panic with len 0.

Reply:
Actually, in apiserver bootstrap, this serverList is also guaranteed not equal to 0.